src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -239,12 +239,42 @@ public static void Copy(Array sourceArray, long sourceIndex, Array destinationAr
             return InternalGetValue(GetFlattenedIndex(new ReadOnlySpan<int>(indices)));
         }
 
-        public object? GetValue(int index)
+        public unsafe object? GetValue(int index)
         {
-            if (Rank != 1)
+            MethodTable* pMethodTable = RuntimeHelpers.GetMethodTable(this);
+
+            // The array rank is 0 for SZ arrays, throw if that's not the case
+            if (pMethodTable->MultiDimensionalArrayRank != 0)
```

Comment:
This is not correct. There are non-SZ arrays with Rank=1. You just cannot create them directly in C#.

Reply:
Ah, yup my bad, forgot you could still have 1D arrays with non-0 lower bounds 😄
Fixed this, and temporarily moved the stuff to the CoreCLR file to hopefully get the CI going while I look into the changes you suggested below. Currently investigating what the CoreCLR equivalent for `RuntimeImports.RhBox` is 🙂